front/lib/resources/space_resource.ts

```diff
@@ -613,7 +687,7 @@ export class SpaceResource extends BaseResource<SpaceModel> {
             { role: "builder", permissions: ["read", "write"] },
             { role: "user", permissions: ["read"] },
           ],
-          groups: this.groups.map((group) => ({
+          groups: this.groups.filter(groupFilter).map((group) => ({
             id: group.id,
             permissions: ["read"],
           })),
```

Comment:
Let's try to avoid quadratic (even though probably not huge)
```suggestion
          groups: this.groups.reduce((acc, group) => {
            if (groupFilter(group)) {
              acc.push({
                id: group.id,
                permissions: ["read"],
              });
            }
            return acc;
          }, [] as GroupPermission[]),
```